### ISSUE TITLE
Define a consistent text treatment for help text underneath form fields

### ISSUE THREAD

**Issue by kjellr** [WordPress/gutenberg#15683]
As brought up in https://github.com/WordPress/gutenberg/pull/15469#discussion_r281265789, Gutenberg is currently inconsistent in its text styling for under-form-field help text. In most cases, it's using italic text, [as defined](https://github.com/WordPress/gutenberg/blob/d82a3bcde825bb23991c2790aad10844169d3f3c/packages/components/src/base-control/style.scss#L18-L21) in the `BaseControl` component: 

<img width="279" alt="Screen Shot 2019-05-16 at 10 39 50 AM" src="https://user-images.githubusercontent.com/1202812/57862992-4ae17e80-77c7-11e9-8a9c-80e4e0fcf63a.png">
<img width="277" alt="Screen Shot 2019-05-16 at 10 38 53 AM" src="https://user-images.githubusercontent.com/1202812/57862993-4ae17e80-77c7-11e9-8f53-78a2ad0872e8.png">
<img width="278" alt="Screen Shot 2019-05-16 at 10 38 29 AM" src="https://user-images.githubusercontent.com/1202812/57862994-4ae17e80-77c7-11e9-95c5-c4db4e888b8d.png">

We have many other cases where similar text is not italic however. In these cases, the text is the exact same weight and size as the form field labels: 

<img width="278" alt="Screen Shot 2019-05-16 at 10 40 17 AM" src="https://user-images.githubusercontent.com/1202812/57863150-8e3bed00-77c7-11e9-944c-562467743252.png">
<img width="280" alt="Screen Shot 2019-05-16 at 10 38 09 AM" src="https://user-images.githubusercontent.com/1202812/57863151-8e3bed00-77c7-11e9-8aa7-666b912596ab.png">

We should settle on a global approach for this sort of help text. As @afercia [pointed out](https://github.com/WordPress/gutenberg/pull/15469#discussion_r281265789), it's important to consider accessibility issues with italic text — among other potential concerns, italic text is known to be more difficult to read for those with dyslexia.

**Comment by kjellr** [comment]
Sharing some quick explorations to fuel discussion: 

---

**Option 1: Remove italic, no other changes**

<img width="277" alt="Screen Shot 2019-05-16 at 11 33 48 AM" src="https://user-images.githubusercontent.com/1202812/57867101-85024e80-77ce-11e9-8b63-ae73e513dac3.png">

<img width="277" alt="Screen Shot 2019-05-16 at 11 41 36 AM" src="https://user-images.githubusercontent.com/1202812/57867686-97c95300-77cf-11e9-9f23-93f59242c2e9.png">


- ➕  Simple, easy to read. 
- ➖  Unless there's a link, text is exactly the same size + style as the field label, which can be confusing if there's another field below this one. 

---

**Option 2: Italic**

<img width="279" alt="Screen Shot 2019-05-16 at 11 40 46 AM" src="https://user-images.githubusercontent.com/1202812/57867633-78cac100-77cf-11e9-8259-2b718684e63f.png">

<img width="276" alt="Screen Shot 2019-05-16 at 11 41 21 AM" src="https://user-images.githubusercontent.com/1202812/57867696-9a2bad00-77cf-11e9-9a71-211091746bca.png">


- ➕  Good differentiation between the help text and the field label.  
- ➖  Italic text is widely regarded to be difficult to read for users with Dyslexia. 

---

**Option 3: Two gray colors**

<img width="278" alt="Screen Shot 2019-05-16 at 11 47 28 AM" src="https://user-images.githubusercontent.com/1202812/57868187-7e74d680-77d0-11e9-9097-ec55e52402ca.png">
<img width="278" alt="Screen Shot 2019-05-16 at 11 48 04 AM" src="https://user-images.githubusercontent.com/1202812/57868188-7e74d680-77d0-11e9-8808-c6e6bdd574b4.png">

This version darkens the field label, and lightens the help text color. I used `$dark-gray-900` and `$dark-gray-300` for now, but we'd certainly refine if we went this direction. 

- ➕  Passable, but not obvious differentiation between the help text and the field label.  
- ❓  There may be unintended consequences to darkening field labels everywhere. 

---

**Option 4: Bold form field labels**

<img width="277" alt="Screen Shot 2019-05-16 at 11 51 19 AM" src="https://user-images.githubusercontent.com/1202812/57868417-f2af7a00-77d0-11e9-9d16-3be94145717e.png">
<img width="277" alt="Screen Shot 2019-05-16 at 11 49 46 AM" src="https://user-images.githubusercontent.com/1202812/57868418-f2af7a00-77d0-11e9-92cc-649343f365dd.png">

- ➕  Good differentiation between the help text and the field label.  
- ➖  We use bold for panel/section headers. Using it for all form fields will remove differentiation there.
- ❓  There are surely other unintended consequences to bolding field labels everywhere. 

---

**Option 6: Font size differences**

<img width="277" alt="Screen Shot 2019-05-16 at 12 02 52 PM" src="https://user-images.githubusercontent.com/1202812/57869315-ce549d00-77d2-11e9-9350-e8f8cfe7916b.png">
<img width="277" alt="Screen Shot 2019-05-16 at 12 03 13 PM" src="https://user-images.githubusercontent.com/1202812/57869316-ce549d00-77d2-11e9-9888-12800ef38e81.png">

Here, I used `11px` labels. Despite being super tiny (to my elder millennial eyes at least), this size is is used in many areas of Gutenberg. 

- ➕  Good differentiation between the help text and the field label.  
- ➖  `11px` text is too small to comfortably read for many users. 


---

**Option 7: Font size differences (with larger text)**

<img width="275" alt="Screen Shot 2019-05-16 at 11 55 17 AM" src="https://user-images.githubusercontent.com/1202812/57868763-9a2cac80-77d1-11e9-9476-7b9d01c5a4bc.png">
<img width="277" alt="Screen Shot 2019-05-16 at 11 55 42 AM" src="https://user-images.githubusercontent.com/1202812/57868764-9a2cac80-77d1-11e9-8197-3c7e4ad51d47.png">


Since I think `11px` is too small, I tried `14px` labels (`1px` larger than they are currently) and `12px` labels (`1px` smaller than they are currently). 

- ➕  Good differentiation between the help text and the field label.  
- ➕  Bigger label text is easier for everyone to read. 
- ➖  Smaller label text is harder for everyone to read.
- ➖  This would be a really complicated change: If we're going to make this text larger, we should definitely make other `13px` text larger too. That'll definitely have repercussions across the interface.

**Comment by melchoyce** [comment]
"Option 3: Two gray colors" looks much better to me than what we have now, just from scanning through the options, and seems to have the fewest negatives. I think this would be the best quick win of the options.

That  said, "Option 7: Font size differences (with larger text)" appeals to me most in the long-term, since our labels are pretty small across the admin — embiggening them everywhere would be a win. I think if we're able to make all our labels 14px (maybe even 15px?), we could potentially leave the help text 13px.

**Comment by mapk** [comment]
Yep, I like **Option 3** the best. While the two grays may not be distinguishable by some, I feel that's okay and the positioning of the help text (under), as opposed to the positioning of the label (top or side), seems to help relate the purpose as well.

Adjusting the font size is something I've heard before and @jasmussen has recently shared some concerns with how small our text is too. Just pinging him for his input too!

**Comment by kjellr** [comment]
> That said, "Option 7: Font size differences (with larger text)" appeals to me most in the long-term, since our labels are pretty small across the admin — embiggening them everywhere would be a win. I think if we're able to make all our labels 14px (maybe even 15px?), we could potentially leave the help text 13px.

One other thing to note about this direction: if we were to up the size of the form field labels, it would probably make sense to up the size of the field text itself. It feels a little weird (to me) to have the actual input text be smaller than the label: 

<img width="276" alt="Screen Shot 2019-05-16 at 1 45 47 PM" src="https://user-images.githubusercontent.com/1202812/57875570-40cc7980-77e1-11e9-8d61-eb87ebbc6e2d.png">
<img width="277" alt="Screen Shot 2019-05-16 at 1 46 59 PM" src="https://user-images.githubusercontent.com/1202812/57875571-40cc7980-77e1-11e9-9778-abf9f5039de1.png">

Also worth noting: on smaller screens, our input field text size is actually bumped up to `16px` already, to avoid browser zoom when focusing on the fields: 

<img width="373" alt="Screen Shot 2019-05-16 at 1 47 36 PM" src="https://user-images.githubusercontent.com/1202812/57875614-5b065780-77e1-11e9-846f-fed42a2fabc8.png">

**Comment by jasmussen** [comment]
Great thoughts here. 

One aspect that I increasingly test when thinking of accessibility in visual design, is to imagine the screen as grayscale. How can I differentiate two blocks of text in that case? This is one of the reasons why underlined text makes good sense.

In that vein, I'm personally mostly drawn to a combination of 4 and 7. 15px feels like too big a text bump unless we also bump font sizes elsewhere (which we should consider, but do in a holistic way). Maybe 14px and semi-bold?

**Comment by jasmussen** [comment]
> Also worth noting: on smaller screens, our input field text size is actually bumped up to 16px already, to avoid browser zoom when focusing on the fields:

It's worth clarifying that this is done because otherwise mobile safari _zooms in_ which is a super jarring effect. There's supposed to be a way to opt out of that zooming, but last I tried it didn't work. This is just to say that this is the mobile safari tax, and not necessarily one we should optimize towards.

**Comment by kjellr** [comment]
> 15px feels like too big a text bump unless we also bump font sizes elsewhere (which we should consider, but do in a holistic way). Maybe 14px and semi-bold?

I agree that 15px feels like too big of a bump. My gut here says that a tiny first step would be to try changing our standard `13px` size to `14px`, and our standard `11px` size to `12px`. That's a small change, but because of the cascading changes this creates, it makes sense to try small changes first. 

Also, I wouldn't want to change this in just the form fields: we'd want to adjust these sizes globally. I wouldn't want to add a `14px` size while keeping a `13px` size too — `1px` font-size increments aren't enough to establish clear hierarchy, and frequently look like mistakes when seen together. It's certainly a big change — but it seems like a `1px` change has to be somewhat manageable... right? 😉 

(All that said, I also feel like a global font size change may be out of the scope of just this one super-specific ticket. 😄)

> This is just to say that this is the mobile safari tax, and not necessarily one we should optimize towards.

I agree that it's definitely a Safari hack, but it's also one that I really like in practice. On mobile, this feels like an excellent font size for these fields.

**Comment by jasmussen** [comment]
I'm very much in favor of bigger font sizes, it's _one_ of the reasons for https://core.trac.wordpress.org/ticket/47012. The toolbars are even designed to be able to accommodate bigger (24x24px) icons.

The reason I'm careful is not lack of desire for better legibility, it's rooted in a past attempt to bump the font size across the board. It was sadly not as simple as changing a variable — there's a ton of invisible complexity in the shifting UI that still has to be able to accommodate translations and mobile viewports. 

To circle back to this ticket specifically, it would _seem_ that bumping the font size of headings, and un-italicizing the help text would be a great first step. Just consider subheadings too.

**Comment by afercia** [comment]
Note: for Core, I've created https://core.trac.wordpress.org/ticket/47327

**Comment by afercia** [comment]
For feedback and interesting resources, please refer to the Trac ticket https://core.trac.wordpress.org/ticket/47327

**Comment by melchoyce** [comment]
@davewhitley has a typography proposal that would cover these styles: https://make.wordpress.org/design/2019/10/11/proposal-a-consistent-type-scale-for-wordpress/

He made a PR to test them out: https://github.com/WordPress/gutenberg/pull/18244

**Comment by davewhitley** [comment]
Yep! Thanks Mel.

Following the typescale:

- Labels would be "Body small"
- Input text would be "Body"
- Help text would be "Caption"

![image](https://user-images.githubusercontent.com/618551/69177220-ba189200-0acc-11ea-9a3e-989d00f142ba.png)

**Comment by ibdz** [comment]
In [Design triage ](https://wordpress.slack.com/archives/C02S78ZAL/p1586880818487100)today, we agreed that the type scale brings a nice balance to the form field. At least settle on the help text should be a point or 2 smaller than the label and a little lighter grey, no italics.

**Comment by rinkalpagdar** [comment]
Currently, It looks fine. Do you have any suggestions here?
<img width="271" alt="Image" src="https://github.com/user-attachments/assets/8115d6fe-1461-48b1-84df-76d2f30c0487" />

**Comment by jasmussen** [comment]
I'd tend to think that this has been addressed! Closing as such, happy to reopen if this was in error!
